components/webpack.common.js

```diff
@@ -0,0 +1,133 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2018 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Creates configurations for webpacking of MathJax3 components
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+const fs = require('fs');
+const path = require('path');
+const webpack = require('webpack');
+
+/**************************************************************/
+
+/**
+ * @param {string} string  The string whose special characters are to be escaped
+ * @return {string}        The string with regex special characters escaped
+ */
+function quoteRE(string) {
+    return string.replace(/([\\.{}[\]()?*^$])/g, '\$1')
+}
+
+/**
+ * Creates the plugin needed for converting mathjax3 references to component/lib references
+ *
+ * @param {string} mathjax3    The location of the mathjax3 js files
+ * @param {string[]} lib       The component library directories to be linked against
+ * @param {string} dir         The directory of the component being built
+ * @return {any[]}             The plugin array (empty or with the conversion plugin)
```

Comment:
What exactly is the conversion plugin?

Reply:
It is the plugin described in the main comment five lines above:  the one that converts `mathjax3` references to `components/lib` references.